vault/vault.go

```diff
@@ -250,6 +359,56 @@ func configureTLS(config *api.Config, secretConfig map[string]interface{}) error
 	return config.ConfigureTLS(&tlsConfig)
 }
 
+func kubernetesAuth(client *api.Client, config map[string]interface{}) (string, error) {
+	authConfig, err := buildAuthConfig(config)
+	if err != nil {
+		return "", err
+	}
+	method, err := kubernetes.NewKubernetesAuthMethod(authConfig)
+	if err != nil {
+		return "", err
+	}
+
+	path, data, err := method.Authenticate(context.TODO(), client)
+	if err != nil {
+		return "", err
+	}
+
+	secret, err := client.Logical().Write(path, data)
```

Comment:
do we need extra permissions now to perform this operation ?

Reply:
No, it doesn't require any additional rules to the portworx policy.